src/test/scala/org/kibanaLoadTest/helpers/KbnClient.scala

```diff
@@ -29,35 +31,39 @@ case class SavedObject(id: String, soType: String)
 
 class KbnClient(config: KibanaConfiguration) {
   private val logger: Logger = LoggerFactory.getLogger("KbnClient")
+  private val MAX_CONCURRENT_CONNECTIONS = 20
   private def getJsonPath(path: Path) =
-    if (path.toString.endsWith(".json")) path else Paths.get(path.toString + ".json")
+    if (path.toString.endsWith(".json")) path
+    else Paths.get(path.toString + ".json")
 
-  def getCookie(): String = {
-    val loginHeaders = Map(
-      "Content-Type" -> "application/json",
-      "kbn-xsrf" -> "xsrf"
-    )
-    val url = config.baseUrl + "/internal/security/login"
-    val loginRequest = new HttpPost(url)
-    loginHeaders foreach {
-      case (key, value) => loginRequest.addHeader(key, value)
-    }
-    val client = HttpClientBuilder.create.build
-    loginRequest.setEntity(new StringEntity(config.loginPayload))
-    val loginResponse: CloseableHttpResponse = null
-    try {
-      val loginResponse = client.execute(loginRequest)
-      loginResponse
-        .getHeaders("set-cookie")(0)
-        .getValue
-        .split(";")(0)
-    } catch {
-      case e: Throwable =>
-        throw new RuntimeException("Login exception", e)
-    } finally {
-      if (loginResponse != null) loginResponse.close()
-      client.close()
+  def getClientAndConnectionManager(
+      perRouteConnections: Int = MAX_CONCURRENT_CONNECTIONS,
+      withAuth: Boolean = true
+  ): (CloseableHttpClient, PoolingHttpClientConnectionManager) = {
+    val connManager = new PoolingHttpClientConnectionManager()
+    connManager.setDefaultMaxPerRoute(perRouteConnections)
+    connManager.setMaxTotal(perRouteConnections)
```

Comment:
It can be confusing to use the same variable for both, so I may change it. The important is to increase `setDefaultMaxPerRoute` which is 2 y default and we are doing 20. `MaxTotal` is 100 by default so it should be fine unless `perRouteConnections > 100`

Reply:
Given how this client is used I think the restriction is fine but maybe it warrants a comment?